edit-post/components/visual-editor/index.js

```diff
@@ -29,7 +28,7 @@ function VisualEditor( { hasFixedToolbar, isLargeViewport } ) {
 			<WritingFlow>
 				<PostTitle />
 				<BlockList
-					showContextualToolbar={ isLargeViewport && ! hasFixedToolbar }
```

Comment:
I think the issue was a failure to preserve the original condition here.

Previously:

`! isMobile( state ) && isFeatureActive( state, 'fixedToolbar' )`

This should have translated to:

`! isLargeViewport || ! hasFixedToolbar`

https://www.wikiwand.com/en/De_Morgan%27s_laws

Reply:
I tried this condition and it's showing two toolbars on small viewports at the same time.

Reply:
>I tried this condition and it's showing two toolbars on small viewports at the same time.

Only because of the other changes in `HeaderToolbar` on this branch.

Apply this diff to master, and it behaves as I'd expect:

```diff
diff --git a/edit-post/components/visual-editor/index.js b/edit-post/components/visual-editor/index.js
index 1d621863a..a439953e3 100644
--- a/edit-post/components/visual-editor/index.js
+++ b/edit-post/components/visual-editor/index.js
@@ -29,7 +29,7 @@ function VisualEditor( { hasFixedToolbar, isLargeViewport } ) {
 			<WritingFlow>
 				<PostTitle />
 				<BlockList
-					showContextualToolbar={ isLargeViewport && ! hasFixedToolbar }
+					showContextualToolbar={ ! isLargeViewport || ! hasFixedToolbar }
 					renderBlockMenu={ ( { children, onClose } ) => (
 						<Fragment>
 							<BlockInspectorButton onClick={ onClose } />
```

Reply:
Gotcha, thanks